### PR TITLE
ENH: `stats.fligner` and `stats.bartlett` empty input handling.

### LINKED ISSUE #689: Statistics Review: bartlett (Trac #162)
_Original ticket http://projects.scipy.org/scipy/ticket/162 on 2006-04-02 by @rkern, assigned to @rkern._

The function `bartlett` in file source:trunk/Lib/stats/morestats.py needs review.

Please look over the StatisticsReview guidelines and add your comments below.

### LINKED ISSUE #693: Statistics Review: fligner (Trac #166)
_Original ticket http://projects.scipy.org/scipy/ticket/166 on 2006-04-02 by @rkern, assigned to @rkern._

The function `fligner` in file source:trunk/Lib/stats/morestats.py needs review.

Please look over the StatisticsReview guidelines and add your comments below.

### PR DESCRIPTION
**Pull request by aeklant**
closes gh-689; closes gh-693

### CODE REVIEW

**Comment by aeklant** [comment]
* `stats.fligner` now returns a `namedtuple`.
* Explicitly check for empty input (returns `(np.nan, np.nan)`).
* Several amendments to documentation (please review for completeness and correctness).

File: scipy/stats/morestats.py [hunk 1]
```diff
@@ -1553,6 +1553,18 @@ def bartlett(*args):
     pvalue : float
         The p-value of the test.
 
+    See Also
+    --------
+    fligner : A non-parametric test for the equality of k variances
+    levene : A parametric test for equality of k variances
```
Comment (josef-pkt) [inline scipy/stats/morestats.py]: I would mention "robust" for levene, as contrast to bartlett
`levene : A robust parametric test for equality of k variances`

what exactly robust means can be read in the docstrings

**Comment by josef-pkt** [comment]
looks fine to me. Did you check test coverage?

**Comment by aeklant** [comment]
@josef-pkt Yes. Test coverage looks good. I will wait to see if I get more comments before I change the `levene` mention.

**Comment by aeklant** [comment]
@rgommers ping

**Comment by rgommers** [comment]
Test failure is timeout, can be ignored.

File: scipy/stats/morestats.py [hunk 2]
```diff
@@ -1560,7 +1572,19 @@ def bartlett(*args):
     .. [2]  Snedecor, George W. and Cochran, William G. (1989), Statistical
               Methods, Eighth Edition, Iowa State University Press.
 
+    .. [3] Park, C. and Lindsay, B. G. (1999). Robust Scale Estimation and
+           Hypothesis Testing based on Quadratic Inference Function. Technical
+           Report #99-03, Center for Likelihood Studies, Pennsylvania State
+           University.
```
Comment (rgommers) [inline scipy/stats/morestats.py]: This only has one citation. On the issue for ``bartlett`` I had suggested to cite the original paper: Bartlett, "Properties of sufficiency and statistical tests"

**Comment by rgommers** [comment]
@aeklant if you can add that one reference, the rest looks good to me. If you push a doc commit with ``[ci skip]`` then it can be merged straight away.

**Comment by aeklant** [comment]
@rgommers done. Please let me know if there are any new comments.

**Comment by rgommers** [comment]
Merged, thanks @aeklant

**Pull request merged**